### DISCUSSION TITLE
frequent https connection issues from action runners since 2024-03-17

### DISCUSSION THREAD

**Discussion by wdiestel** [community/community#117648] [Actions]
### Select Topic Area

General

### Body

I've got connection issues in around 10% of my actions steps towards a server I get additional config info from. It started 2014-03-17 together with a runner image update of ubuntu-22.04. Connecting to the server from other places does not show any problems, so it is somehow on the network path from Github actions to the target server. It persists within one action step - so later retry does not help, in other words somehow it is sticky to an action runner.

What could I do towards finding the root cause or getting any improvement here? Any suggestions?

Thanks, 
  Wolfram.

~~~
Linux fv-az658-803 6.5.0-1017-azure #17~22.04.1-Ubuntu SMP Sat Mar  9 04:50:38 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    RX:  bytes packets errors dropped  missed   mcast           
         16.8k     126      0       0       0       0 
    TX:  bytes packets errors dropped carrier collsns           
         16.8k     126      0       0       0       0 
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP mode DEFAULT group default qlen 1000
    link/ether 00:22:48:0a:ae:fe brd ff:ff:ff:ff:ff:ff
    RX:  bytes packets errors dropped  missed   mcast           
         12.0M   9.32k      0       0       0       0 
    TX:  bytes packets errors dropped carrier collsns           
          483k   1.49k      0       0       0       0 
3: docker0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN mode DEFAULT group default 
    link/ether 02:42:13:c4:9a:a1 brd ff:ff:ff:ff:ff:ff
    RX:  bytes packets errors dropped  missed   mcast           
             0       0      0       0       0       0 
    TX:  bytes packets errors dropped carrier collsns           
             0       0      0       0       0       0 
# ...
curl: (28) Connection timed out after 20000 milliseconds    
# ...
State    Recv-Q Send-Q Local Address:Port    Peer Address:Port Process                     
LISTEN   0      500          0.0.0.0:8084         0.0.0.0:*                                
LISTEN   0      128          0.0.0.0:ssh          0.0.0.0:*                                
LISTEN   0      4096   127.0.0.53%lo:domain       0.0.0.0:*                                
ESTAB    0      0          10.1.0.62:38714  20.150.90.196:https timer:(keepalive,4.212ms,0)
ESTAB    0      0          10.1.0.62:38722  20.150.90.196:https timer:(keepalive,4.208ms,0)
SYN-SENT 0      1          10.1.0.62:41610  <TARGET IP>:https timer:(on,992ms,1)         
~~~

**Comment by the-gsk** [comment]
It seems like you're experiencing intermittent connection issues from your GitHub Actions runners to a specific server, which started occurring after a runner image update on March 17th, 2024. Here are some steps you can take to troubleshoot and potentially resolve the issue:

1. **Check Network Configuration**: Ensure that there are no misconfigurations in your network settings, both on the GitHub Actions side and on the server side. Make sure that the necessary ports are open and that there are no firewall rules blocking the connections.

2. **Verify DNS Resolution**: Confirm that DNS resolution is working correctly on your GitHub Actions runners. If DNS resolution is failing, it could cause connection timeouts.

3. **Review Server Logs**: Check the logs on the target server to see if there are any indications of network issues or errors. This might provide insights into what is causing the connection timeouts.

4. **Monitor Network Traffic**: Use network monitoring tools to analyze the network traffic between your GitHub Actions runners and the target server. This can help identify any patterns or anomalies that could be causing the connection timeouts.

5. **Review GitHub Status**: Check the GitHub Status page to see if there are any ongoing issues or maintenance that could be affecting the connectivity of GitHub Actions runners.

6. **Contact GitHub Support**: If you're unable to resolve the issue on your own, consider reaching out to GitHub Support for assistance. Provide them with as much information as possible, including any error messages or logs you've collected during your troubleshooting efforts.

By following these steps, you should be able to identify the root cause of the connection issues and take appropriate actions to resolve them.

**Comment by wdiestel** [comment]
Hi Gaurav,
thank you for your hints. I think I already checked there what I could check from my side:
- (1,2) Well I checked everything I could on the client side (i.e. action runner) - DNS works fine, ip a/l looks basically ok.
- For those failed connections DNS name is resolved to the correct IP adress, but seems the initial SYN tcp packet is not answered; may be dropped by some firewall?
- (3) Server side is just web space provider - I can only see into access.log where those failed connections are not visible
- (4) I have no access to network components in between the client and the server
- (5) The issue persisted over several weeks, it is not related to Github status in general though
- (6) The support page at https://support.github.com/request doesn't show me a category "network issues" or "github actions", so I'm not sure what is the correct way to reach out to the support - may be I could use the category "feedback"?
